crates/database/src/orders.rs

```diff
@@ -651,29 +671,44 @@ pub struct FeeUpdate {
     pub surplus_fee: Option<BigDecimal>,
     pub surplus_fee_timestamp: DateTime<Utc>,
     pub full_fee_amount: BigDecimal,
+    pub sell_token: Address,
+    pub buy_token: Address,
+    pub sell_amount: BigDecimal,
 }
 
+/// Updates the `surplus_fee` of multiple orders and returns their `uid`s.
 pub async fn update_limit_order_fees(
     ex: &mut PgConnection,
-    order_uid: &OrderUid,
     update: &FeeUpdate,
-) -> Result<(), sqlx::Error> {
+) -> Result<Vec<OrderUid>, sqlx::Error> {
     const QUERY: &str = "
         UPDATE orders
         SET
             surplus_fee = $1,
             surplus_fee_timestamp = $2,
             full_fee_amount = $3
-        WHERE uid = $4
+        WHERE
+            sell_token = $4
+            AND buy_token = $5
+            AND sell_amount = $6
+        RETURNING
+            uid
     ";
-    sqlx::query(QUERY)
+    #[derive(FromRow)]
+    struct OrderUidRow {
+        uid: OrderUid,
+    }
+    let rows: Vec<OrderUidRow> = sqlx::query_as(QUERY)
```

Comment:
If somebody knows a way to make this nicer let me know. 🙏

Reply:
I think you can use `query_scalar` (instead of `query_as`) to get SQLx to fetch a vector of UIDs and not need the intermediate row type.

I'm not the expert though, and not 100% sure this will work.

Reply:
Thanks that worked: [5feb574](https://github.com/cowprotocol/services/pull/1037/commits/5feb574d533b10dac8c9ab6c01a77d3e0b3490b6).

Reply:
Intuitively a `HASH` index would make more sense but postgres doesn't support multi column hash indices so I reverted the commit again.